osu.Game/Graphics/Containers/Markdown/OsuMarkdownContainer.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using Markdig;
+using Markdig.Extensions.AutoIdentifiers;
+using Markdig.Extensions.Tables;
+using Markdig.Extensions.Yaml;
+using Markdig.Syntax;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Containers.Markdown;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Graphics.Sprites;
+
+namespace osu.Game.Graphics.Containers.Markdown
+{
+    public class OsuMarkdownContainer : MarkdownContainer
```

Comment:
Actually after checking in osu!web, the margin bottom of the paragraph is 1.5em.

- https://github.com/ppy/osu-web/blob/376cac43a051b9c85ce95e2c446099be187b3e45/resources/assets/less/bem/osu-md.less#L230
- https://github.com/ppy/osu-web/blob/376cac43a051b9c85ce95e2c446099be187b3e45/resources/assets/less/variables.less#L58

Because the base font size 14px, so the actual margin bottom is 1.5 * 14 = 21px.

I crosscheck with devtools and the actual margin bottom is correct 21px.

![Screenshot from 2021-05-06 09-08-06](https://user-images.githubusercontent.com/4964985/117232082-ae4c9700-ae4a-11eb-8eb7-8f0f023a0654.png)

I know line space and margin is a bit different, but I try to compare paragraph in osu!web with line spacing 10 and 21.

| osu!web | spacing 10 | spacing 21 |
| --- | --- | --- |
| ![Screenshot from 2021-05-06 09-16-41](https://user-images.githubusercontent.com/4964985/117232791-1a7bca80-ae4c-11eb-94a6-0792b923c891.png) | ![Screenshot from 2021-05-06 09-15-38](https://user-images.githubusercontent.com/4964985/117232804-236c9c00-ae4c-11eb-912b-ef19660ae428.png) | ![Screenshot from 2021-05-06 09-13-50](https://user-images.githubusercontent.com/4964985/117232813-28c9e680-ae4c-11eb-8d59-f13c57b53f71.png) |

After some trial and error, I get value between 17 or 18.

| osu!web | spacing 18 | spacing 17 |
| --- | --- | --- |
| ![Screenshot from 2021-05-06 09-16-41](https://user-images.githubusercontent.com/4964985/117232791-1a7bca80-ae4c-11eb-94a6-0792b923c891.png) | ![Screenshot from 2021-05-06 09-22-31](https://user-images.githubusercontent.com/4964985/117233069-9c6bf380-ae4c-11eb-99aa-e56d77d0f889.png) | ![Screenshot from 2021-05-06 09-23-39](https://user-images.githubusercontent.com/4964985/117233228-e7860680-ae4c-11eb-9c09-982d2db2d27a.png) |

I don't know if this spacing is different on another screen resolution.

Reply:
These numbers should not be trial-and-error, though. The metrics should be transferred directly from osu-web (as they are done in the user profile overlay and other common implementations).

Reply:
Yes, I propose to use 21px value from margin bottom value in osu-web as line spacing here. But as you can see in first table comparison, the spacing is a bit different between **osu-web** and **spacing 21**.

Reply:
That's fine. The numbers should match, even if visually it looks different.